store/src/prune_list.rs

```diff
@@ -54,16 +55,21 @@ pub struct PruneList {
 
 impl PruneList {
 	/// Instantiate a new prune list from the provided path and bitmap.
-	pub fn new(path: Option<PathBuf>, mut bitmap: Bitmap) -> PruneList {
-		// Note: prune list is 1-indexed so remove any 0 value for safety.
-		bitmap.remove(0);
-
-		PruneList {
+	/// Note: Does not flush the bitmap to disk. Caller is responsible for doing this.
+	pub fn new(path: Option<PathBuf>, bitmap: Bitmap) -> PruneList {
+		let mut prune_list = PruneList {
 			path,
-			bitmap,
+			bitmap: Bitmap::create(),
 			shift_cache: vec![],
 			leaf_shift_cache: vec![],
+		};
+
+		for pos in bitmap.iter().filter(|x| *x > 0) {
```

Comment:
the non-zero test is either wrong (for 0-based pos) or redundant (for 1-based pos).

Reply:
agreed in theory, but removing this test breaks existing assumptions about the underlying data